scripts/release.sh

```diff
@@ -59,43 +58,169 @@ function release_app {
+    if [[ $1 == $2 ]]
+    then
+        return 0
+    fi
+    local IFS=.
+    local i ver1=($1) ver2=($2)
+    # fill empty fields in ver1 with zeros
+    for ((i=${#ver1[@]}; i<${#ver2[@]}; i++))
+    do
+        ver1[i]=0
+    done
+    for ((i=0; i<${#ver1[@]}; i++))
+    do
+        if [[ -z ${ver2[i]} ]]
+        then
+            # fill empty fields in ver2 with zeros
+            ver2[i]=0
+        fi
+        if ((10#${ver1[i]} > 10#${ver2[i]}))
+        then
+            return 1
+        fi
+        if ((10#${ver1[i]} < 10#${ver2[i]}))
+        then
+            return 2
+        fi
+    done
+    return 0
+}
+
+function git_branch_commit_and_release() {
+    local BRANCH_OR_TAG_VALUE=$2
+    local REMOTE=origin COMMIT_BRANCH=$TRAVIS_BRANCH
+    if [[ $GITHUB_TOKEN ]]; then
+        REMOTE=https://$GITHUB_TOKEN@github.com/$TRAVIS_REPO_SLUG
+    else
+        echo "Missing environment variable GITHUB_TOKEN=[GitHub Personal Access Token]"
+        exit 1
+    fi
+    version_compare $1 $2
+    COMPARE=$?
+    if [[ ${COMPARE} = 1 ]]
+    then
+        echo "VERSION value" $1 "is greater than" $3 "version" $2
+    elif [[ ${COMPARE} = 2 ]]
+    then
+        echo "VERSION value" $1 "is less than" $3 "version" $2
+    fi
+
+    for (( p=`grep -o "\."<<<".$BRANCH_OR_TAG_VALUE"|wc -l`; p<3; p++)); do 
+        BRANCH_OR_TAG_VALUE+=.0;
+        done;
+    echo "Setting VERSION to " ${BRANCH_OR_TAG_VALUE}
+    
+    if [[ $3 = "tag" ]];
+    then
+        git fetch ${REMOTE} $TRAVIS_BRANCH
+        git branch $TRAVIS_BRANCH FETCH_HEAD
+        COMMIT_BRANCH=HEAD
+        increment_version $BRANCH_OR_TAG_VALUE 3
+        BRANCH_OR_TAG_VALUE=$TAG_INCREASED_VERSION
+        echo "Version incremented to ${BRANCH_OR_TAG_VALUE}"
+    fi
+
+    git checkout $TRAVIS_BRANCH
+    echo ${BRANCH_OR_TAG_VALUE} > VERSION
+    git add VERSION
+    # make Travis CI skip this build
+    COMMIT_MESSAGE="chore: Update VERSION file to ${BRANCH_OR_TAG_VALUE} [ci skip]"
+    git commit -m "${COMMIT_MESSAGE}"
+    if ! git push --quiet --follow-tags ${REMOTE} ${COMMIT_BRANCH} > /dev/null 2>&1; then
+        echo "Failed to push git changes to" $TRAVIS_BRANCH
+        exit 1
+    fi
+    if [ ! -z $4 ]; then
+        VERSION=${BRANCH_OR_TAG_VALUE}-$4
+    fi
+    echo "Starting version" ${VERSION} "release"
+    release_process
+
+    # Update develop VERSION value to match the latest released version
+    git fetch ${REMOTE} develop
+    git branch develop FETCH_HEAD
+    git checkout develop
+    DEV_VERSION=`cat VERSION`
+    version_compare ${DEV_VERSION} ${BRANCH_OR_TAG_VALUE}
+    COMPARE=$?
+    if [[ ${COMPARE} = 2 ]]
+    then
+        echo "Updating develop branch version to " ${BRANCH_OR_TAG_VALUE}
+        echo ${BRANCH_OR_TAG_VALUE} > VERSION
+        git add VERSION
+        git commit -m "${COMMIT_MESSAGE}" #Skip travis build on develop commit
+        git push ${REMOTE} develop
+    else
+        echo "Develop branch VERSION value is not updated"
```

Comment:
Do you want to use here the word `VERSION`or the env var `$VERSION`?

Reply:
Word VERSION referring to the VERSION file